---
title: 'Final Project: Icterid Tree'
author: "Nathan Burroughs"
date: "3/19/2015"
output: html_document
---

I have a document of Icterid data documententing many variables, including nest shape and plumage. The purpose was to analyze wether there is a correlation between nest shape and sexual dichromatism, which would indicate natural selection acting on the females, giving the females cryptic plumage. This study involved a phylogenetic tree which we gathered from previous papers. For this project, I am making my own phylogenetic tree for the family Icteridae. The following are my steps:

1. I saved the document as a comma deliminated file Icteridnestshape.txt, and uploaded it into my virtual machine. 

2. Then I wrote a python code using regular expressions to take out each genus and species:

```{python}
python FC_Speciesname.py
```
  
This gave me the file genus_and_species.txt

3. Next I created and ran a python file to scrape the Genbank database for all genome sequences that matched Icteridae and Cytochrome B

```{python}
python FC_getgenes3.py
```

This gave me the fasta file icterid.fasta

4. This file wasn't perfect; it had duplicates and some full mitochondrial genomes. So I had to remove these:

```{python}
python FC_justCYTBnd.py
```    
Which gave me the file icteridcytnd.fasta

5. I had to align this file, and I had the clustalw program downloaded from biopython, so I ran:

```{unix}
clustalw
```    
And then I chose the input file: FC_justCYTBnd.py, chose multiple sequence alignment, and saved it as the nexus file icteridcytnd.nex, and saved it again as test3.nex.


6. This file wasn't perfect for running Mr. Bayes on, because of the 'gi' line. I changed this to just the species and genus by opening and saving test3.nex with the following program:

```{python}
python FC_fix.py
```

7. With the aligned nexus file, I was ready to run Mr. Bayes. I edited test3.nex into test5.nex in order to make it compatible with Mr. Bayes and then I ran Mr. Bayes on test5.nex:

```{python}
python FC_Bayes_icterids.py
```    
This gave me the file test5.nex.con.tre

8. I wrote R script in the R file:

```{unix}
generating_trees.r
```    
And got four PDFs of phylogenetic trees:

1. Icteridtree1.pdf - Complete tree
2. Icteridtree2.pdf - Complete tree with Agelaius Phoeniceus removed (something weird about it)
3. Icteridtree3.pdf - Fan Tree
4. Icteridtree4.pdf - Slightly smaller font

All these files are on my Github.

Anyways, that's how I got my phylogenetic trees!